dask/dataframe/io/parquet.py

```diff
@@ -309,6 +309,8 @@ def _read_fastparquet(fs, fs_token, paths, columns=None, filters=None,
                        categories, pf.schema, pf.cats, pf.dtypes,
                        pf.file_scheme, storage_name_mapping)
            for i, rg in enumerate(rgs)}
+    for catcol in pf.cats:
+        meta[catcol] = meta[catcol].cat.set_categories(pf.cats[catcol])
```

Comment:
I don't think this works if `columns` isn't a superset of `partition_on`, as `meta` wouldn't contain those column names originally (e.g. if I wrote a file with `partition_on=['a', 'b']`, and read with `columns=['b']`, this would error as `'a'` isn't in the metadata, but is in `pf.cats`).

If my intuition is correct, then a test should be expanded to hit a combination of `partition_on` and `columns` that would have triggered this error, as currently this code passed our test suite.

Reply:
You may be right, I'll check. I guess the partitioned column could also end up as the index?